File: backend/schemas/ai.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class AIChatRequest(BaseModel):
    message: str = Field(min_length=1, max_length=1200)
    wedding_plan_id: int | None = None


class AIRecommendationItem(BaseModel):
    id: int | None = None
    package_id: int
    provider_id: int
    category: str
    provider: str
    package: str
    price: float
    reason: str
    rating: float | None = None
    location: str | None = None
    capacity: int | None = None


class AIRecommendationOption(BaseModel):
    title: str
    total_estimated_cost: float
    remaining_budget: float | None = None
    difference: str
    items: list[AIRecommendationItem] = Field(default_factory=list)


class AIRecommendationSummary(BaseModel):
    total_estimated_cost: float
    remaining_budget: float | None = None
    recommendation_summary: str | None = None
    items: list[AIRecommendationItem] = Field(default_factory=list)
    options: list[AIRecommendationOption] = Field(default_factory=list)


class AIReplacementData(BaseModel):
    category: str
    provider_id: int
    package_id: int
    provider: str
    package: str
    price: float
    reason: str
    rating: float | None = None
    location: str | None = None
    capacity: int | None = None


class AIReplacementResponse(BaseModel):
    success: bool
    message: str
    data: AIReplacementData | None = None


class AIChatResponse(AIRecommendationSummary):
    conversation_id: int
    message: str


class AIStoredMessage(BaseModel):
    id: int
    sender: str
    message: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class AIHistoryResponse(BaseModel):
    conversation_id: int
    messages: list[AIStoredMessage] = Field(default_factory=list)


class AISavedRecommendation(BaseModel):
    id: int
    customer_id: int
    wedding_plan_id: int
    total_estimated_cost: float
    remaining_budget: float | None = None
    recommendation_summary: str | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/ai.ts
export type AIRecommendationItem = {
  id?: number | null;
  package_id: number;
  provider_id: number;
  category: string;
  provider: string;
  package: string;
  price: number;
  reason: string;
  rating?: number | null;
  location?: string | null;
  capacity?: number | null;
};

export type AIRecommendationSummary = {
  total_estimated_cost: number;
  remaining_budget?: number | null;
  recommendation_summary?: string | null;
  items: AIRecommendationItem[];
  options?: AIRecommendationOption[];
};

export type AIReplacementData = AIRecommendationItem;

export type AIReplacementResponse = {
  success: boolean;
  message: string;
  data?: AIReplacementData | null;
};

export type AIRecommendationOption = {
  title: string;
  total_estimated_cost: number;
  remaining_budget?: number | null;
  difference: string;
  items: AIRecommendationItem[];
};

export type AIChatResponse = AIRecommendationSummary & {
  conversation_id: number;
  message: string;
};

export type ChatMessage = {
  id: string;
  sender: "user" | "assistant";
  message: string;
};

export type AIStoredMessage = {
  id: number;
  sender: ChatMessage["sender"];
  message: string;
  created_at: string;
};

export type AIHistoryResponse = {
  conversation_id: number;
  messages: AIStoredMessage[];
};
